std/wasi/snapshot_preview1_test.ts

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable */
+
+import { assert, assertEquals } from "../testing/asserts.ts";
+import * as path from "../path/mod.ts";
+import WASI from "./snapshot_preview1.ts";
+
+if (import.meta.main) {
+  const options = JSON.parse(Deno.args[0]);
+  const binary = await Deno.readFile(Deno.args[1]);
+  const module = await WebAssembly.compile(binary);
+
+  const wasi = new WASI({
+    env: options.env,
+    args: options.args,
+    preopens: options.preopens,
+  });
+
+  const instance = new WebAssembly.Instance(module, {
+    wasi_snapshot_preview1: wasi.exports,
+  });
+
+  wasi.memory = instance.exports.memory;
+
+  instance.exports._start();
+} else {
+  const rootdir = path.dirname(path.fromFileUrl(import.meta.url));
+  const testdir = path.join(rootdir, "testdata");
+  const outdir = path.join(testdir, "snapshot_preview1");
+
+  for await (const entry of Deno.readDir(testdir)) {
+    if (!entry.name.endsWith(".rs")) {
+      continue;
+    }
+
+    const process = Deno.run({
+      cmd: [
+        "rustc",
+        "--target",
+        "wasm32-wasi",
+        "--out-dir",
+        outdir,
+        path.join(testdir, entry.name),
+      ],
+      stdout: "inherit",
+      stderr: "inherit",
+    });
+
+    const status = await process.status();
+    assert(status.success);
+
+    process.close();
+
+    // TODO(caspervonb) allow the prelude to span multiple lines
+    const source = await Deno.readTextFile(path.join(testdir, entry.name));
+    const prelude = source.match(/^\/\/\s*\{.*/);
+    if (prelude) {
+      const basename = entry.name.replace(/.rs$/, ".json");
+      await Deno.writeTextFile(
+        path.join(outdir, basename),
+        prelude[0].slice(2)
+      );
+    }
+  }
+
+  for await (const entry of Deno.readDir(outdir)) {
+    if (!entry.name.endsWith(".wasm")) {
+      continue;
+    }
+
+    Deno.test(entry.name, async function () {
+      const basename = entry.name.replace(/\.wasm$/, ".json");
+      const prelude = await Deno.readTextFile(path.resolve(outdir, basename));
+      const options = JSON.parse(prelude);
+
+      const process = await Deno.run({
+        cwd: testdir,
+        cmd: [
+          `${Deno.execPath()}`,
+          "run",
+          "--quiet",
+          "--unstable",
+          "--v8-flags=--experimental-wasm-bigint",
+          "--allow-all",
+          import.meta.url,
+          prelude,
+          path.resolve(outdir, entry.name),
+        ],
+        stdin: "piped",
+        stdout: "piped",
+        stderr: "piped",
+      });
+
+      if (process.stdin) {
```

Comment:
What's this conditional for? Shouldn't process.stdin always be present?

Reply:
Regression in the 1.1 type definitions; can't get at it without a type guard.

Reply:
https://github.com/denoland/deno/issues/6270

Reply:
Actually, faulty memory in action here; I had a ternary in the run options to `"inherit" : "piped"`, went for always piped because #6270 made it always null.

So this is just left-over cruft from that, should be removed yes.